content/en/docs/guides/deadlines.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Deadlines
+description: >-
+  Explains how deadlines can be used to effectively deal with unreliable
+  backends.
+---
+
+### Overview
+
+A deadline is used to specify how long a client is willing to wait for a server
+to respond. This simple idea is very important in building robust distributed
+systems. Clients that do not wait around unnecessarily and servers that know
+when to give up processing requests will improve the resource utilization and
+latency of your system.
+
+Note that while some language APIs have the concept of a __deadline__, others
+use the idea of a __timeout__. When an API asks for a deadline, you provide a
+point in time which the request should not go past. A timeout is the max
+duration of time that the request can take. For simplicity, we will only refer
+to deadline in this document.
+
+### Deadlines on the Client
+
+The default deadline for a gRPC client is some very large number that would not
+be appropriate to use in any real world scenario. This means you should always
+explicitly set a realistic deadline in your clients. To determine the
+appropriate deadline you would ideally start with an educated guess based on
+what you know about your system (network latency, server processing time, etc.),
+validated by some load testing.
+
+If a server has gone past the deadline when processing a request, the client
+will give up and fail the RPC with the `DEADLINE_EXCEEDED` status. Note that the
+client giving up on a particular RPC does not mean that the server will
+stop processing the request.
+
+### Deadlines on the Server
+
+A gRPC server might receive RPCs from a client with an unrealistically short
+deadline that would not give the server enough time to ever respond in time.
+This would result in the server just wasting valuable resources and in a worst
+case scenario, crash the server. To avoid situations like these, the server
+should always check the deadline provided in the request and stop processing a
+request once it knows the client has stopped waiting for the response.
+
+If a server is not able to respond in time, it should free up any resources it
+was using to process the request and respond with a `CANCELLED` status.
+
+#### Deadline Propagation
```

Comment:
This section could use a diagram to illustrate:

- conversion of deadline to timeout (and vice versa) in the processing over wire
- propagation through multiple servers (client -> server1 -> server2)